packages/kbn-tinymath/test/library.test.js

```diff
@@ -31,96 +43,134 @@ describe('Parser', () => {
+      expect(parse('"-foo0 bar1"')).toEqual(variableEqual('-foo0 bar1'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
     });
   });
 
   describe('Functions', () => {
     it('no arguments', () => {
-      expect(parse('foo()')).toEqual({ name: 'foo', args: [] });
+      expect(parse('foo()')).toEqual(functionEqual('foo', []));
     });
 
     it('arguments', () => {
-      expect(parse('foo(5,10)')).toEqual({ name: 'foo', args: [5, 10] });
+      expect(parse('foo(5,10)')).toEqual(functionEqual('foo', [5, 10]));
     });
 
     it('arguments with strings', () => {
-      expect(parse('foo("string with spaces")')).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
+      expect(parse('foo("string with spaces")')).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
 
-      /* eslint-disable prettier/prettier */
-      expect(parse("foo('string with spaces')")).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
-      /* eslint-enable prettier/prettier */
+      expect(parse("foo('string with spaces')")).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
+    });
+
+    it('named only', () => {
+      expect(parse('foo(q=10)')).toEqual(functionEqual('foo', [namedArgumentEqual('q', '10')]));
+    });
+
+    it('named and positional', () => {
+      expect(parse('foo(ref, q="bar")')).toEqual(
+        functionEqual('foo', [variableEqual('ref'), namedArgumentEqual('q', 'bar')])
+      );
+    });
+
+    it('numerically named', () => {
+      expect(() => parse('foo(1=2)')).toThrow('but "(" found');
+    });
+
+    it('multiple named', () => {
+      expect(parse('foo(q_param="bar", offset-type="1d")')).toEqual(
+        functionEqual('foo', [
+          namedArgumentEqual('q_param', 'bar'),
+          namedArgumentEqual('offset-type', '1d'),
+        ])
+      );
+    });
+
+    it('multiple named and positional', () => {
+      expect(parse('foo(q="bar", ref, offset="1d", 100)')).toEqual(
+        functionEqual('foo', [
+          namedArgumentEqual('q', 'bar'),
+          variableEqual('ref'),
+          namedArgumentEqual('offset', '1d'),
+          100,
+        ])
+      );
+    });
+
+    it('duplicate named', () => {
+      expect(parse('foo(q="bar", q="test")')).toEqual(
+        functionEqual('foo', [namedArgumentEqual('q', 'bar'), namedArgumentEqual('q', 'test')])
+      );
+    });
+
+    it('incomplete named', () => {
+      expect(() => parse('foo(a=)')).toThrow('but "(" found');
+      expect(() => parse('foo(=a)')).toThrow('but "(" found');
```

Comment:
Is it possible to use a sub-expression in a named argument (`foo(a=abc("def"))`)? I don't think we absolutely need it, but either way a test would be cool.

Judging from the pegjs grammar I would say we are limited to strings here